### ISSUE TITLE
Infer return types based on inlay type hints

### ISSUE THREAD

**Issue by egorio** [rehearsal-js/rehearsal-js#543]
TS provides functions to get [Inlay Hints](https://code.visualstudio.com/docs/typescript/typescript-editing).

Those hints include:
- FunctionParameterTypeHints
- includeInlayVariableTypeHints
- includeInlayPropertyDeclarationTypeHints
- includeInlayFunctionLikeReturnTypeHints
- includeInlayEnumMemberValueHints

We can utilize them to add types where TS didn't provide CodeFixes.

For example, the next code doesn't trigger suggestions for many places where inline has types:

![image](https://user-images.githubusercontent.com/1502496/208182895-d203ed6b-60a9-466f-91db-a3f20bcdb024.png)

Ref:
- LanguageService.[provideInlayHints](https://github.com/microsoft/TypeScript/blob/3716ffe748fe799b1c69090056397bda814c0d49/src/services/types.ts#L596)()

Sister to: https://github.com/rehearsal-js/rehearsal-js/issues/526

**Comment by lynchbomb** [comment]
This is excellent! especially for return types.

**Comment by wondersloth** [comment]
Found that this would benefit the migrate case from our internal testing with `urn-utils`.

Related to #696 

Prioritizing `Object` from a JSDoc blocks the inferred return type. 

@egorio and I validated that this inlay hints has the same output as when you use `Refactor` from the right click menu in the editor. 
![refactor-inlay-type-hints](https://user-images.githubusercontent.com/129781/216729109-f1b6217a-eb33-476e-8a44-4fe909cbbcc3.gif)
